fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryChooserPlanBase.java

```diff
@@ -141,27 +138,11 @@ public boolean isStrictlySorted() {
      * that, in all, when the same "selected index" is chosen for all picker value, one would get back a consistent set
      * of values, representing one of the child plans for this plan.
      *
-     * @return list of {@link ValuePickerValue} representing all the values from all the sub plans
+     * @return a {@link ValuePickerValue} representing the values from all the sub plans
      */
-    private static List<? extends Value> calculateChildrenValues(@Nonnull final List<? extends Quantifier> quantifiers) {
-        // Store all values in a multimap, indexed by the ordinal of the value in the returned list
-        // Each list represents all the i'th Value from each of the sub plans
-        ImmutableListMultimap.Builder<Integer, Value> mapBuilder = ImmutableListMultimap.builder();
-        quantifiers.forEach(quantifier -> {
-            List<? extends Value> values = quantifier.getFlowedValues();
-            for (int i = 0 ; i < values.size() ; i++) {
-                mapBuilder.put(i, values.get(i));
-            }
-        });
-        ImmutableListMultimap<Integer, ? extends Value> valuesMap = mapBuilder.build();
-
-        ImmutableList.Builder<ValuePickerValue> resultBuilder = ImmutableList.builder();
-        for (int i = 0 ; i < valuesMap.keySet().size() ; i++) {
-            ImmutableList<? extends Value> subValues = valuesMap.get(i);
-            // For now, fix all the picker values to return the first sub value
-            resultBuilder.add(new ValuePickerValue(0, subValues));
-        }
-
-        return resultBuilder.build();
+    private static Value calculateChildrenValues(@Nonnull final List<? extends Quantifier> quantifiers) {
+        return new ValuePickerValue(0, quantifiers.stream()
```

Comment:
It looks like the `ValuePickerValue` is set to choose the first qun's result type, is that intentional?

Reply:
It was like that before I touched it -- Ohad knows more about this.